src/rgw/driver/sfs/types.h

```diff
@@ -357,20 +396,34 @@ class Bucket {
     return info.flags;
   }
 
-  ObjectRef get_or_create(const rgw_obj_key &key);
-
-  ObjectRef get(const std::string &name) {
-    auto it = objects.find(name);
-    if (it == objects.end()) {
-      throw UnknownObjectException();
-    }
-    return objects[name];
-  }
-
-  void finish(const DoutPrefixProvider *dpp, const std::string &objname);
+ private:
+  bool want_new_version(const rgw_obj_key &key, ObjectRef obj);
 
+ public:
+  // Return object ref for key
```

Comment:
same as before; I think it would be helpful to have these as docstrings instead of generic comments.